### PR TITLE
feat: Add support for Rich Text

### LINKED ISSUE #1307: Implement rich text editor for long descriptions in the app
### Feature Request

**Feature Description**

Description fields need to support fields like bullet points and special symbols, for which a rich text editor library need to be implemented.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by mishuvs**
Fixes #1307 

Checklist:

- [ ] I have checked for PMD and check-style issues 
- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream development branch.

Changes: Adds support for rich text

Screenshots for the change:
videotogif_2018 08 18_20 41 48

### CODE REVIEW

File: app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java [hunk 1]
```diff
@@ -103,6 +107,12 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
             }
         });
 
+        binding.form.description.setOnClickListener(view -> {
+            Intent i = new Intent(getContext(), RichEditorActivity.class);
+            i.putExtra(TAG_RICH_TEXT, binding.form.description.getText().toString());
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/core/event/create/UpdateEventFragment.java]: No single letter variable names

File: app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+import com.eventyay.organizer.databinding.TextEditorLayoutBinding;
+
+public class RichEditorActivity extends AppCompatActivity {
+
+    public static final String TAG_RICH_TEXT = "rich_text";
+
+    TextEditorLayoutBinding binding;
+    AlertDialog linkDialog;
+    AlertDialog saveAlertDialog;
+    String description;
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java]: Why not private

File: app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+        ActionBar actionBar = getSupportActionBar();
+        if (actionBar != null) {
+            actionBar.setHomeButtonEnabled(true);
+            actionBar.setDisplayHomeAsUpEnabled(true);
+        }
+
+        binding.editor.setPlaceholder(getString(R.string.enter_text));
+
+        Intent i = getIntent();
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java]: No single letter variable names

File: app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        binding.actionInsertNumbers.setOnClickListener(v -> binding.editor.setNumbers());
+
+        binding.actionInsertLink.setOnClickListener(v -> {
+
+            if (linkDialog == null) {
+                createLinkDialog();
+            }
+
+            linkDialog.show();
+
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java]: Too many extra lines

**Comment by mishuvs** [comment]
I have made the changes @iamareebjamal @Masquerade0097 @sridharjajoo @srv-twry Please review

File: app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java [hunk 5]
```diff
@@ -0,0 +1,137 @@
...
+        }
+        return super.onOptionsItemSelected(item);
+    }
+
+    @Override
+    public void onBackPressed() {
+        if (binding.editor.getHtml().equals(description)) {
+            saveChanges(description);
+            return;
+        } else if (saveAlertDialog == null) {
```
Comment (iamareebjamal) [inline app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java]: Remove the else if. It should be a standalone if condition

File: app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            case R.id.action_save:
+                saveChanges(binding.editor.getHtml());
+                break;
+            default:
```
Comment (srv-twry) [inline app/src/main/java/com/eventyay/organizer/ui/editor/RichEditorActivity.java]: Usually we add a comment saying 'nothing here' or equivalent :)

**Pull request merged**